### DISCUSSION TITLE
is there any way to modify existing object?

### DISCUSSION THREAD

**Discussion by klys-equinix** [hayes/pothos#1225] [General]
We generate our objects using Prisma. We then expose those objects using Pothos. 
We are using Federation, and we need to add `shareable: true` to some of the fields.
Is there any way to add an attribute to a field of existing object?

**Comment by hayes** [comment]
Can you share the code your using to define the fields?

Is there a reason you need to define shareable separately?  There are probably ways to do this, but they would likely be pretty hacky.  The easiest is probably to build the schema first, then add the directives to the built schema (this isn't Pothos specific)

**Reply by klys-equinix** [reply]
The pothos federation plugin specifies that the directives have to be added with field definition:
https://pothos-graphql.dev/docs/plugins/federation#field-directives-directives

but in my case those definitions are generated from prisma.

I found a way to do it which is pretty hacky - redefine the object while copying the generated code:

```
import { CagesObject, CagesCabinetsFieldObject } from '@inventory-graphql/pothos/dist/generated/Cages/object.base.mjs';

export function generateCagesObject() {
  builder.prismaObject('Cages', {
    ...CagesObject,
    fields: (t: any) => ({
      ...CagesObject.fields(t),
      cabinets: t.relation('cabinets', {
        ...CagesCabinetsFieldObject(t),
        shareable: true
      })
    })
  });
}
````
